lib/ngx/process.lua

```diff
@@ -0,0 +1,53 @@
+-- Copyright (C) Yichun Zhang (agentzh)
+
+
+local ffi = require 'ffi'
+local base = require "resty.core.base"
+local errmsg = base.get_errmsg_ptr()
+local FFI_ERROR = base.FFI_ERROR
+local ffi_str = ffi.string
+local ngx_phase = ngx.get_phase
+
+
+local process_type_name = {
```

Comment:
We should use plural form of words in container variable names. Here this definitely holds all possible names, so the identifier should be `process_type_names` instead.